content/influxdb/cloud-serverless/process-data/send-alerts.md

```diff
@@ -0,0 +1,328 @@
+2.  Use the queried data and tools available in your runtime to send alerts.
+
+---
+
+## Create a Slack app
+
+To send alerts to Slack, first create a Slack app and gather the required
+connection credentials to interact with your app.
+More information is provided in the
+[Slack basic app setup documentation](https://api.slack.com/authentication/basics).
+
+## Install dependencies
+
+{{% note %}}
+This guide assumes you have already
+[setup your Python project and virtual environment](/influxdb/cloud-serverless/query-data/sql/execute-queries/python/#create-a-python-virtual-environment).
+{{% /note %}}
+
+Use `pip` to install the following dependencies:
+
+- `influxdb_client_3`
+- `pandas`
+- `slack_sdk`
+
+```sh
+pip install influxdb3-python pandas slack_sdk
+```
+
+## Create an InfluxDB client
+
+Use the `InfluxDBClient3` function in the `influxdb_client_3` module to 
+instantiate an InfluxDB client.
+Provide the following credentials:
+
+- **host**: [{{< cloud-name >}} region URL](/influxdb/cloud-serverless/reference/regions)
+  _(without the protocol)_
+- **org**: InfluxDB organization name
+- **token**: [InfluxDB API token](/influxdb/cloud-serverless/admin/tokens/) with
+  read and write permissions on the bucket you want to query and write to.
+- **database**: InfluxDB bucket name
+
+{{% code-placeholders "(API|BUCKET|ORG)_(NAME|TOKEN)" %}}
+```py
+from influxdb_client_3 import InfluxDBClient3
+import pandas
+
+# Instantiate an InfluxDBClient3 client configured for your bucket
+influxdb = InfluxDBClient3(
+    host='cloud2.influxdata.com',
+    org='ORG_NAME',
+    token='API_TOKEN',
+    database='BUCKET_NAME'
+)
+```
+{{% /code-placeholders %}}
+
+## Create a Slack client
+
+1.  Import the `WebClient` function from the `slack.sdk` module and the `SlackApiError`
+    function from the `slack_sdk.errors` module.
+2.  Use the `WebClient` function to instantiate a Slack client.
+    Provide the following credentials:
+
+    - **token**: [Slack bot token](https://api.slack.com/authentication/basics#getting-your-authentication-token)
+
+{{% code-placeholders "SLACK_BOT_TOKEN" %}}
+```py
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+
+slack = WebClient(token='SLACK_BOT_TOKEN')
+```
+{{% /code-placeholders %}}
+
+## Query InfluxDB
+
+Define either a SQL or InfluxQL query to retrieve data to alert on.
+Depending on what data you want to alert on, you can:
+
+- Include logic in the query so it only returns results that should be alerted on.
+- Query data necessary for further processing and then send alerts based on 
+  processing performed in your runtime.
+
+The example query below only returns values above a threshold that should trigger alerts.
+
+{{< code-tabs-wrapper >}}
+{{% code-tabs %}}
+[SQL](#)
+[InfluxQL](#)
+{{% /code-tabs %}}
+
+<!--------------------------------- BEGIN SQL --------------------------------->
+{{% code-tab-content %}}
+```sql
+SELECT
+  time,
```

Comment:
In this example you just specify time where in a later example:
` selector_last(co, time)['time'] AS time,`

Are both considered valid?

Reply:
Oh, thank you!